plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hadoop/TrinoHadoopCatalog.java

```diff
@@ -0,0 +1,317 @@
+    {
+        return session.getQueryId();
+    }
+
+    /**
+     * Convert a session to a properties map that is used to initialize a new catalog
+     * together with other catalog properties configured at connector level.
+     * The default behavior is to return an empty map.
+     * Implementations can override this method to pass session properties or session identity information to the catalog.
+     * @param session session
+     * @return catalog properties derived from session
+     */
+    public Map<String, String> getSessionProperties(ConnectorSession session)
+    {
+        return EMPTY_SESSION_MAP;
+    }
+
+    private Catalog createNewCatalog(ConnectorSession session)
+    {
+        ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
+        builder.putAll(catalogProperties);
+        builder.putAll(getSessionProperties(session));
+        Configuration hadoopConf = hdfsEnvironment.getConfiguration(new HdfsEnvironment.HdfsContext(session), new Path(warehouse));
+        return loadCatalog(CATALOG_IMPL, catalogName.toString(), builder.buildOrThrow(), hadoopConf);
+    }
+
+    public Catalog getCatalog(ConnectorSession session)
+    {
+        try {
+            return catalogCache.get(getSessionCacheKey(session), () -> createNewCatalog(session));
+        }
+        catch (ExecutionException e) {
+            throw new IllegalStateException("Fail to create catalog for " + session, e);
+        }
+    }
+
+    private SupportsNamespaces getNamespaceCatalog(ConnectorSession session)
+    {
+        Catalog catalog = getCatalog(session);
+        if (catalog instanceof SupportsNamespaces) {
+            return (SupportsNamespaces) catalog;
+        }
+        throw new TrinoException(NOT_SUPPORTED, "catalog " + CATALOG_IMPL + " does not support namespace operations");
+    }
+
+    private Map<String, String> convertToCatalogProperties(IcebergConfig config)
+    {
+        return ImmutableMap.of(WAREHOUSE_LOCATION, config.getCatalogWarehouse());
+    }
+
+    private TableIdentifier toTableId(SchemaTableName schemaTableName)
+    {
+        return TableIdentifier.of(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+    }
+
+    private SchemaTableName schemaFromTableId(TableIdentifier tableIdentifier)
+    {
+        return new SchemaTableName(tableIdentifier.namespace().toString(), tableIdentifier.name());
+    }
+
+    @Override
+    public List<String> listNamespaces(ConnectorSession session)
+    {
+        return getNamespaceCatalog(session).listNamespaces().stream().map((namespace) -> {
+            return namespace.toString();
+        }).collect(Collectors.toList());
+    }
+
+    @Override
+    public void dropNamespace(ConnectorSession session, String namespace)
+    {
+        SupportsNamespaces catalog = getNamespaceCatalog(session);
+        catalog.dropNamespace(catalog.listNamespaces().stream().filter(_namespace -> _namespace.toString() == namespace).findAny().orElse(null));
```

Comment:
We should be using `.equals()` instead of `==` here
```suggestion
        catalog.dropNamespace(catalog.listNamespaces().stream().filter(_namespace -> namespace.equals(_namespace.toString())).findAny().orElse(null));
```